worker/uniter/runner/context/context.go

```diff
@@ -384,6 +387,19 @@ func (ctx *HookContext) ResetExecutionSetUnitStatus() {
 	ctx.hasRunStatusSet = false
 }
 
+// PrincipalUnit will return the principal unit name and set the environment
+// variable JUJU_PRINCIPAL_UNIT.
+func (ctx *HookContext) PrincipalUnit() (string, error) {
```

Comment:
the principal unit doesn't change, so it would be a bit neater to have this method as:

```go
func (ctx *HookContext) PrincipalUnit() string {
    return ctx.principal
}
```

and then in the `updateContext` method, set it:

```go
principal, err := ctx.unit.PrincipalUnit()
if err != nil {
    return err
}
ctx.principal = principal
```

... but it occurs to me that the HookContext.PrincipalUnit method isn't even needed at the moment. It's not used by any hook tools (config-get, relation-get, etc.). So you can probably just drop it.